### ISSUE TITLE
predicate `array?` and `each` throwing an error

### ISSUE THREAD

**Issue by mrbongiolo** [dry-rb/dry-validation#170]
When using:
```ruby
required(:skills).filled(:array?)
```

It throws that error:
```
/home/mrbongiolo/.gem/ruby/2.3.0/gems/dry-types-0.7.2/lib/dry/types/compiler.rb:15:in `visit': undefined method `[]' for nil:NilClass (NoMethodError)
```

Also, when using `each` like this:
```ruby
required(:skills).each(:str?)
```

And calling the schema with: `{ skills: "some string" }` it throws this error:
```
 undefined method `map' for "some string":String
```

And I can't use `:array?` before `each` also because of the error above...

This is using the `master` branch.

**Comment by solnic** [comment]
`each` adds `array?` automatically, that's odd. I'll look into it.

**Comment by mrbongiolo** [comment]
I was trying to reproduce the error that I'm getting with `required(:skills).filled(:array?)` on the dry-validation specs, but couldn't make it.

Probably something is not right in my environment, I'll try to figure out was is messing with it, if I find anything will update here.

**Comment by mrbongiolo** [comment]
Ok I found how to reproduce it.

This is happening when setting the input_processor to `:sanitizer` or `:form`.

**Comment by mrbongiolo** [comment]
This is the falling AST generated by the Schema when using those input_processor.

```ruby
ast = [
      :type, [
        "hash", [
          :schema, [
            [:key, [:skills, [:type, "array"]]]
          ]
        ]
      ]
    ]
```

**Comment by mrbongiolo** [comment]
Fixed on latest release, specs on #172 are passing.
